File: rectangulo.cpp
#include <iostream>
using namespace std;
class recta{
    public:
        int ancho;
        int alto;
    public:
        recta(int,int);
        void CalcularArea();
        void CalcularPerimetro();
        void CambiarAlto();
        void CambiarAncho();
};
recta::recta(int _ancho, int _alto){
    ancho= _ancho;
    alto= _alto;
}
void recta::CalcularArea(){
    float x= ancho*alto;
    cout<<"\n el Area es "<<x<<""<<endl;
}
void recta::CalcularPerimetro(){
    float x= 2*ancho+2*alto;
    cout<<"\n el perimetro es "<<x<<""<<endl;
}
void recta::CambiarAlto(){
    string d;
    cout<<"\n quiere cambiar alto y(si) o n(no)", cin>> d;
    if (d=="y")
        {
            int newalto;
            cout<<"\n ingrese numero ", cin>> newalto ;
            alto= newalto;
        }
    else{
            cout<<"\n no cambia alto";
        }
}
void recta::CambiarAncho(){
    string d;
    cout<<"\n quiere cambiar ancho y(si) o n(no)", cin>> d;
    if (d=="y")
        {
            int newancho;
            cout<<"\n ingrese numero ", cin>> newancho ;
            ancho= newancho;
        }
    else{
            cout<<"\n no cambia ancho";
        }
}
int main()
    {
        string a;
        recta c1=recta(2, 5);
        cout<<"\n desea continuar y(si) o n(no)",cin>> a;
        while (a=="y"){
            c1.CambiarAlto();
            c1.CambiarAncho();
            c1.CalcularArea();
            c1.CalcularPerimetro();
            cout<<"\n Desea continuar y(si) o n(no)",cin>> a;
        }
    };

File: tarea semana 3/rectangulo.py
class rectangulo:
    def __init__(self,ancho,alto):
        self.ancho= ancho
        self.alto=alto
    def cambiar_ancho(self,_ancho):
        self.ancho=_ancho
    def cambiar_alto(self,_alto):
        self.alto=_alto
    def calcularArea(self):
        x=self.alto*self.ancho
        print("El area es: ", x)
    def calcularPeri(self):
        x=2*self.alto+2*self.ancho
        print("El perimetro es: ", x)
rec1= rectangulo(3,4)
l=1
while l>0:
    v=str(input("Desea cambiar alto y(si) o n(no) "))
    if v=="y":
        rec1.cambiar_alto(int(input("Ingrese alto: ")))
    b=str(input("Desea cambiar ancho y(si) o n(no) "))
    if b=="y":
         rec1.cambiar_ancho(int(input("Ingrese ancho: ")))
    rec1.calcularArea()
    rec1.calcularPeri()
    t=str(input("ingrese y para continuar o n para salir : "))
    if t=="n":
        l=0